eth-contracts/contracts/service/ServiceProviderFactory.sol

```diff
@@ -33,20 +33,10 @@ contract ServiceProviderFactory is RegistryContract {
     );
```

Comment:
We should be validating serviceType throughout against the whitelist of two possible inputs

Don't need to implement this yet - but should also consider how we would attach different min/max stake values to different node types, and whether this structure would allow for that extension

Reply:
yes this is a good point, currently you can register _anything_ as an endpoint

Reply:
i like the idea of having a whitelisted min/max amount for each type, the main q would be the tracking aspect of this and how total stake maps to account vs service

Reply:
cool - no need to build it now - just if we can sketch out whether that is possible I would be happy with us having the path to do that in future

The validation would probably require switching to a loop on nodes registered to address and summing up mins/maxes as needed but it seems easily possible

Reply:
Here's an example of how i think the enforcement would look like (not including a whitelist of known service types)- 

Enforced type based min stake:

cnode: min stake, 10 AUD, max stake 200 AUD
disc prov: min stake: 5 AUD, max stake 150 AUD
x type: min stake, max stake

Imagine an account 0x100 with 100 AUD:
Account 0x100 -> register disc prov, stake 6 AUD -> Account MIN STAKE = 5 AUD for 1 disc pov -> Approved!

Account 0x100 -> register 2nd disc prov, stake 0 AUD -> Account MIN STAKE = 10 AUD for 2 disc provs -> Rejected, as total final stake (6)

Account 0x100 -> register 2nd disc prov, stake 20 AUD -> Account MIN STAKE = 10 AUD for 2 disc provs -> Approved, total final stake (26)

Account 0x100 -> register cnode, stake 0 AUD -> Acct MIN STAKE = 20 (2 disc provs, 1 cnode) -> Approved, total final stake (26)

Account 0x100 -> register 2nd cnode, stake 0 AUD -> Acct MIN STAKE = 30 (2 disc provs, 2 cnode) -> Rejected, total final stake (26)

Account 0x100 -> register 2nd cnode, stake 10 AUD -> Acct MIN STAKE = 30 (2 disc provs, 2 cnode) -> Approved, total final stake (36)

2 disc provs, 2 cnodes, min stake = 30

Enforce the min/max bounds on any operations manipulating the stake amount for this account based on the currently registered numbers